src/main/kotlin/no/nav/syfo/varsel/arbeidstaker/ArbeidstakerVarselApi.kt

```diff
@@ -74,12 +74,14 @@ fun Route.registerArbeidstakerVarselApi(
                 val hasAccessToVarsel = motedeltakerArbeidstaker.personIdent == requestPersonIdent
                 if (hasAccessToVarsel) {
                     if (referat != null) {
-                        dialogmotedeltakerService.lesReferatArbeidstaker(
-                            personIdentNumber = requestPersonIdent,
-                            dialogmotedeltakerArbeidstakerUuid = motedeltakerArbeidstaker.uuid,
-                            referatUuid = varselUuid,
-                        )
-                    } else {
+                        if (referat.lestDatoArbeidstaker == null) {
```

Comment:
Slå sammen de 2 nøstede if'ene til 1 om det ikke skal være noe annen logikk her?

Reply:
He, he - gjorde det først, men da virker det ikke: blir NullPointerException i else-delen.

Reply:
Hmm, rart. Jeg kan ta en titt evt. Ser det er noen !!-sjekker på objekt som allerede er null-sjekket, så tror ting kan forenkles litt her.

Reply:
Hmm. Jeg føler det blir lettere å lese om koden blir mer av typen:
1. If Varsel exists -> les varsel
2. or else If Referat exists -> les Varsel
3. or else throw exception.

Må gjenbruke et par linjer kode for å få det til, men til gjengeld så slipper man !!-sjekker og så synes jeg det blir lettere å følge flyten.

Reply:
Prøver å refaktorere.